### PR TITLE
documentation

### PR DESCRIPTION
**Pull request by Quadrisheriff**
i have added code samples to elements, components and layout but not yet created their full documentation. Will continue working on it and hopefully send more PR soon

### CODE REVIEW

**Comment by MrCreeper1008** [comment]
Thank you for your contribution!

**Comment by Quadrisheriff** [comment]
the PR was not succesfull

**Comment by MrCreeper1008** [comment]
yea i think its because you used some links with `http` protocol instead of `https`.

**Comment by Quadrisheriff** [comment]
Ok, will edit that and create a new PR.

**Comment by MrCreeper1008** [comment]
You don't need to create a new PR. Just push new commits into the branch you are working on, and they'll appear here.

File: documentation/docs/components/layout/hero.md [hunk 1]
```diff
@@ -3,5 +3,314 @@ id: hero
...
+  
+
+### **Vertical alignment**
+
+
+<div>
+ <Section>
+  <Hero size="fullheight" color="primary" >
+    <Hero.Head renderAs="header">
+        <div className="bd-notification is-info">Header</div>
```
Comment (MrCreeper1008) [inline documentation/docs/components/layout/hero.md:283]: Remove `bd-notification`

File: documentation/docs/components/layout/hero.md [hunk 2]
```diff
@@ -3,5 +3,314 @@ id: hero
...
+    </Hero.Head>
+    <Hero.Body>
+        Body
+    </Hero.Body>
+    <Hero.Footer>
+        <div className="bd-notification is-danger">Footer</div>
```
Comment (MrCreeper1008) [inline documentation/docs/components/layout/hero.md:289]: Same here.

File: documentation/docs/components/layout/hero.md [hunk 3]
```diff
@@ -3,5 +3,314 @@ id: hero
...
+  </Hero>
+  </Section>
+</div>
+
+
+
+```jsx
+  <Hero size="fullheight" color="primary" >
+    <Hero.Head renderAs="header">
+        <div className="bd-notification is-info">Header</div>
```
Comment (MrCreeper1008) [inline documentation/docs/components/layout/hero.md:300]: And here.

File: documentation/docs/components/layout/hero.md [hunk 4]
```diff
@@ -3,5 +3,314 @@ id: hero
...
+    </Hero.Head>
+    <Hero.Body>
+        Body
+    </Hero.Body>
+    <Hero.Footer>
+        <div className="bd-notification is-danger">Footer</div>
```
Comment (MrCreeper1008) [inline documentation/docs/components/layout/hero.md:306]: Here as well.

File: documentation/docs/components/components/menu.md [hunk 5]
```diff
@@ -4,4 +4,131 @@ title: Menu
...
+import Menu from 'react-bulma-components/lib/components/menu';
+import Box from 'react-bulma-components/lib/components/box';
+import Section from 'react-bulma-components/lib/components/section';
+
+---
+
+### **Import**
+
+
+```shell
```
Comment (MrCreeper1008) [inline documentation/docs/components/components/menu.md:16]: Language should be `js` instead of `shell`.

**Comment by Quadrisheriff** [comment]
I  have added the changes that you requested for, I also created more documentation by adding code samples into column elements, and variables into all layout elements.

How will I correct these errors so that you can review all the changes

File: documentation/docs/components/columns/gap.md [hunk 6]
```diff
@@ -4,5 +4,106 @@ title: Gap
...
+  <Columns.Column size={3}>
+    <Hero color="light">Size 1</Hero>
+  </Columns.Column>
+  <Columns.Column size={3}>
+    <Hero color="white">Size 1</Hero>
+  </Columns.Column>
+  <Columns.Column size={3}>
+    <Hero color="danger">Size 1</Hero>
+  </Columns.Column>
+</Columns>
```
Comment (MrCreeper1008) [inline documentation/docs/components/columns/gap.md:45]: Please indent with 2 spaces. Also, the `varableGap` prop was indented with 2 extra spaces. Please remove them.

File: documentation/docs/components/columns/gap.md [hunk 7]
```diff
@@ -4,5 +4,106 @@ title: Gap
...
+```jsx
+<Columns
+      variableGap={{
+        mobile: 1,
+        tablet: 0,
+        desktop: 3,
+        widescreen: 8,
+        fullhd: 2,
+      }}
+    >
```
Comment (MrCreeper1008) [inline documentation/docs/components/columns/gap.md:57]: Inconsistent indent here.

File: documentation/docs/components/columns/gap.md [hunk 8]
```diff
@@ -4,5 +4,106 @@ title: Gap
...
+  <Columns.Column>
+     <Hero color="success">First Column</Hero>
+  </Columns.Column>
+  <Columns.Column>
+     <Hero color="info">Middle Column</Hero>
+  </Columns.Column>
+  <Columns.Column>
+     <Hero color="warning">Last Column</Hero>
+  </Columns.Column>
+</Columns>
```
Comment (MrCreeper1008) [inline documentation/docs/components/columns/gap.md:88]: Indentation

File: documentation/docs/components/columns/nesting.md [hunk 9]
```diff
@@ -4,5 +4,84 @@ title: Nesting
...
+          </Columns.Column>
+          <Columns.Column size="one-quarter">
+          <Hero color="warning">Second Nested Column</Hero>
+          </Columns.Column>
+          <Columns.Column>
+          <Hero color="warning">Thirld Nested Column</Hero>
+          </Columns.Column>
+      </Columns>
+  </Columns.Column>
+</Columns>
```
Comment (MrCreeper1008) [inline documentation/docs/components/columns/nesting.md:49]: Indentation

File: documentation/docs/components/columns/options.md [hunk 10]
```diff
@@ -4,5 +4,58 @@ title: Options
...
+  </Columns.Column>
+  <Columns.Column>
+    <Hero color="primary">
+      <p>
+      Second column with more content. This is so you can see the vertical
+      alignment.
+      </p>
+    </Hero>
+  </Columns.Column>
+</Columns>
```
Comment (MrCreeper1008) [inline documentation/docs/components/columns/options.md:36]: Indentation

File: documentation/docs/components/columns/responsiveness.md [hunk 11]
```diff
@@ -4,5 +4,230 @@ title: Responsiveness
...
+      </Columns.Column>
+      <Columns.Column>
+        <Hero color="warning">Third column</Hero>
+      </Columns.Column>
+      <Columns.Column>
+        <Hero color="warning">Fourth column</Hero>
+      </Columns.Column>
+    </Columns>
+  </Box>
+</div>
```
Comment (MrCreeper1008) [inline documentation/docs/components/columns/responsiveness.md:84]: Indentation

File: documentation/docs/components/columns/responsiveness.md [hunk 12]
```diff
@@ -4,5 +4,230 @@ title: Responsiveness
...
+          </p>
+        </Hero>
+      </Columns.Column>
+      <Columns.Column className="column">
+        <Hero color="info">1</Hero>
+      </Columns.Column>
+      <Columns.Column className="column">
+        <Hero color="warning">1</Hero>
+      </Columns.Column>
+</Columns>
```
Comment (MrCreeper1008) [inline documentation/docs/components/columns/responsiveness.md:184]: Indentation.

File: documentation/docs/components/columns/responsiveness.md [hunk 13]
```diff
@@ -4,5 +4,230 @@ title: Responsiveness
...
```
Comment (MrCreeper1008) [inline documentation/docs/components/columns/responsiveness.md:223]: Indentation

File: documentation/docs/components/components/card.md [hunk 14]
```diff
@@ -50,6 +48,29 @@ The card container comprises of different elements.
 ## **Examples**
 
 <Section>
+<<<<<<< HEAD
```
Comment (MrCreeper1008) [inline documentation/docs/components/components/card.md:51]: This indicates a git merge conflict. You need to resolve this one and other conflicts before pushing the changes.

File: documentation/docs/components/elements/content.md [hunk 15]
```diff
@@ -26,6 +24,99 @@ import Content from 'react-bulma-components/lib/components/content';
 ## **Examples**
 
 <Section>
+<<<<<<< HEAD
```
Comment (MrCreeper1008) [inline documentation/docs/components/elements/content.md:27]: Please fix merge conflicts in this file.

**Comment by Quadrisheriff** [comment]
I have reindented the code samples and solved all merge conflict issues, the table formatting will come with my next PR.

File: documentation/docs/components/elements/box.md [hunk 16]
```diff
@@ -114,22 +64,106 @@ import Box from 'react-bulma-components/lib/components/box';
...
+      display: {
+        value: 'inline-block',
+        },
+      },
+    }}
+  hide={{
+    tablet: {
+      hide: true,
+      only: true,
+      },
```
Comment (MrCreeper1008) [inline documentation/docs/components/elements/box.md:98]: two extra spaces here; `}` should align with `t`

File: documentation/docs/components/elements/box.md [hunk 17]
```diff
@@ -114,22 +64,106 @@ import Box from 'react-bulma-components/lib/components/box';
...
+    widescreen: {
+      hide: true,
+      },
```
Comment (MrCreeper1008) [inline documentation/docs/components/elements/box.md:101]: `}` should align with `w`

File: documentation/docs/components/elements/box.md [hunk 18]
```diff
@@ -114,22 +64,106 @@ import Box from 'react-bulma-components/lib/components/box';
...
->>>>>>> 334610c6b32e76faf67525d23be6a5b1d27abe51
+</Box>
+```
+
+<Box      
+  responsive={{
+    mobile: {
+      display: {
+        value: 'block'
+        },
```
Comment (MrCreeper1008) [inline documentation/docs/components/elements/box.md:75]: `}` should match with `r`

File: documentation/docs/components/elements/box.md [hunk 19]
```diff
@@ -114,22 +64,106 @@ import Box from 'react-bulma-components/lib/components/box';
...
+      },
+    tablet: {
+      display: {
+        value: 'flex'
+        },
+      },
```
Comment (MrCreeper1008) [inline documentation/docs/components/elements/box.md:81]: `}` should match with `t`

File: documentation/docs/components/elements/box.md [hunk 20]
```diff
@@ -114,22 +64,106 @@ import Box from 'react-bulma-components/lib/components/box';
...
+  responsive={{
+    mobile: {
+      display: {
+        value: 'block'
+        },
+      },
+    tablet: {
+      display: {
+        value: 'flex'
+        },
```
Comment (MrCreeper1008) [inline documentation/docs/components/elements/box.md:80]: `}` should match with `d`.

File: documentation/docs/components/elements/box.md [hunk 21]
```diff
@@ -114,22 +64,106 @@ import Box from 'react-bulma-components/lib/components/box';
...
+<Box      
+  responsive={{
+    mobile: {
+      display: {
+        value: 'block'
+        },
+      },
+    tablet: {
+      display: {
+        value: 'flex'
```
Comment (MrCreeper1008) [inline documentation/docs/components/elements/box.md:79]: missing a trailing comma

File: documentation/docs/components/elements/box.md [hunk 22]
```diff
@@ -114,22 +64,106 @@ import Box from 'react-bulma-components/lib/components/box';
...
-      </Media>
->>>>>>> 334610c6b32e76faf67525d23be6a5b1d27abe51
+</Box>
+```
+
+<Box      
+  responsive={{
+    mobile: {
+      display: {
+        value: 'block'
```
Comment (MrCreeper1008) [inline documentation/docs/components/elements/box.md:74]: missing a trailing comma

File: documentation/docs/components/elements/box.md [hunk 23]
```diff
@@ -114,22 +64,106 @@ import Box from 'react-bulma-components/lib/components/box';
...
+    desktop: {
+      display: {
+        value: 'inline-flex',
+        only: true,
+        },
+      },
+    widescreen: {
+      display: {
+        value: 'inline-block',
+        },
```
Comment (MrCreeper1008) [inline documentation/docs/components/elements/box.md:91]: `}` should match with `d`

File: documentation/docs/components/elements/box.md [hunk 24]
```diff
@@ -114,22 +64,106 @@ import Box from 'react-bulma-components/lib/components/box';
...
+      },
```
Comment (MrCreeper1008) [inline documentation/docs/components/elements/box.md:92]: `}` should match with `w`

File: documentation/docs/components/elements/box.md [hunk 25]
```diff
@@ -114,22 +64,106 @@ import Box from 'react-bulma-components/lib/components/box';
...
+    }}
+  hide={{
+    tablet: {
+      hide: true,
+      only: true,
+      },
+    widescreen: {
+      hide: true,
+      },
+    }}>
```
Comment (MrCreeper1008) [inline documentation/docs/components/elements/box.md:152]: `>` goes to a new line

File: documentation/docs/components/elements/icon.md [hunk 26]
```diff
@@ -37,10 +37,12 @@ import Icon from 'react-bulma-components/lib/components/icon';
 
 
 ```jsx
-<div>
-  <Icon icon="bars" color="info" />
-  <Icon icon="angle-down" />
-</div>
+<Section>
+ <div>
```
Comment (MrCreeper1008) [inline documentation/docs/components/elements/icon.md:41]: need one extra space

File: documentation/docs/components/elements/notification.md [hunk 27]
```diff
@@ -27,60 +27,67 @@ import Notification from 'react-bulma-components/lib/components/notification';
...
 
 ```jsx
-<Notification>
-   Lorem ipsum dolor sit amet, consectetur adipiscing elit lorem ipsum dolor. <strong>Pellentesque risus mi</strong>, tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum <a href="/">felis venenatis</a> efficitur. Sit amet, consectetur adipiscing elit
-   <Button remove />x
-</Notification>
+<Section>
+  <Notification>
+    Lorem ipsum dolor sit amet, consectetur adipiscing elit lorem ipsum dolor. <strong>Pellentesque risus mi</strong>, tempus quis placerat ut, porta nec nulla. Vestibulum rhoncus ac ex sit amet fringilla. Nullam gravida purus diam, et dictum <a href="/">felis venenatis</a> efficitur. Sit amet,consectetur adipiscing elit
+    <Button remove />x
```
Comment (MrCreeper1008) [inline documentation/docs/components/elements/notification.md:41]: Should this be `<Button remove>x</Button>` instead? correct me if im wrong.

**Pull request merged**